sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -192,7 +192,7 @@ def _on_event_received(self, partition_context, event):
                 partition_context._last_received_event = (  # pylint: disable=protected-access
                     event
                 )
-            self._handle_callback(self._event_handler, partition_context, event)
+            self._event_handler(partition_context, event)
```

Comment:
Should we remove method EventProcessor._handle_callback() and call those callbacks directly like in async?

Reply:
Does it trigger on_error if _event_handler raises an exception?

Reply:
updated, async and sync codes are consistent now